addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -226,19 +218,27 @@ public void setStatusInfo(ThingStatus status, ThingStatusDetail statusDetail, St
 
     @Override
     public FritzahaWebInterface getWebInterface() {
-        return this.connection;
+        if (getThing().getThingTypeUID().equals(PL546E_STANDALONE_THING_TYPE)) {
+            return connection;
+        } else {
+            Bridge bridge = getBridge();
+            if (bridge != null) {
+                BridgeHandler handler = bridge.getHandler();
+                if (handler != null && handler instanceof BoxHandler) {
+                    return ((BoxHandler) handler).getWebInterface();
+                }
+            }
+        }
+        return null;
```

Comment:
@martinvw Yes, I like your idea. Do you agree that I may do that in a follow up PR. I have another PR in the pipeline which splits up the `BoxHandler` and `DeviceHandler` into three handlers, introducing a new one for FRITZ! Powerline devices. It will fit better into that PR.